DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -520,7 +525,18 @@ public DMASTStatement Statement(bool requireDelimiter = true)
                     statement = ProcStatement();
                     if (statement != null) {
                         Whitespace();
-                        procStatements.Add(statement);
+                        switch(statement)
+                        {
+                            case DMASTAggregate<DMASTProcStatementSet> greg:
+                                setStatements.AddRange(greg.Statements);
+                                break;
+                            case DMASTProcStatementSet set:
+                                setStatements.Add(set);
+                                break;
+                            default:
+                                procStatements.Add(statement);
+                                break;
+                        }
```

Comment:
There's no need to check for an aggregate here if you handle it in DMProcBuilder
```suggestion
                        if (statement.Is<DMASTProcStatementSet>()) {
                            setStatements.Add(statement);
                        } else {
                            procStatements.Add(statement);
                        }
```

(This is assuming my previous suggestion for `Is<T>()`)